karrot/groups/serializers.py

```diff
@@ -280,6 +318,41 @@ def update(self, instance, validated_data):
         return instance
 
 
+class TrustActionSerializer(serializers.ModelSerializer):
+    """Responsible for both creating and destroying trust"""
+    class Meta:
+        model = GroupMembership
+        fields = ('role', )
+
+    # default so it works with non-trust-for-role aware frontend
+    role = serializers.CharField(default=GROUP_EDITOR)
+
+    def validate_role(self, role):
+        group = self.instance.group
+        if role not in group.all_role_names:
```

Comment:
I guess this is where we can check if the person giving the trust has sufficient role to give that trust (i.e. only editor (or editor and approved) should be able to give trust for approved role - as per discussion in frontend PR). _and/or_ configurable in the `Role` model....